service_contracts/src/FilecoinWarmStorageService.sol

```
@@ -598,7 +598,44 @@ contract FilecoinWarmStorageService is
         // Verify the client's signature
         verifyDeleteDataSetSignature(payer, info.clientDataSetId, signature);
 
-        // TODO Data set deletion logic
+        // Check if the data set is terminated
```

@Chaitu-Tatipamula The current epoch should be greater than the end epoch of both the `pdpEndEpoch ` and `cdnEndEpoch ` as well in addition to them being non-0. Otherwise let's throw an informative error to the user saying "please wait for the payment rail to finalise at epoch X before deleting the dataset".

---

Done @aarshkshah1992